pallets/pips/src/lib.rs

```diff
@@ -1163,6 +1160,18 @@ decl_module! {
 }
 
 impl<T: Trait> Module<T> {
+    fn config<SV, X, E>(origin: T::Origin, new: X, event: E) -> DispatchResult
+    where
+        SV: StorageValue<X, Query = X>,
+        X: FullCodec + Clone,
```

Comment:
does it offers more in comparison to `Encode + Decode`?

Reply:
Yep, https://substrate.dev/rustdocs/v2.0.0/parity_scale_codec/trait.FullCodec.html; `trait StorageValue<T>` requires `T: FullEncode`. Now I'm not sure exactly why `StorageValue` even has a `T` parameter, but it does. :)